app/main/helpers/diff_tools.py

```diff
@@ -0,0 +1,275 @@
+import difflib
+from abc import ABCMeta, abstractmethod
+from flask import Markup
```

Comment:
Since `Markup` objects are rendered as-is they open up a possibility for an XSS or HTML injection attack if they include any user input. For diffs, that means that if the new service version contains any HTML tags they will be rendered in the output.

To prevent this we need to either call `flask.escape` manually on any part of the user data that gets added to the `Markup` objects.